src/megolm/ratchet.rs

```diff
@@ -208,22 +208,21 @@ pub(super) struct RatchetPickle {
 
 impl From<Ratchet> for RatchetPickle {
     fn from(ratchet: Ratchet) -> Self {
-        Self { key: ratchet.inner.into(), counter: ratchet.counter }
+        Self { key: (*ratchet.inner).into(), counter: ratchet.counter }
     }
 }
 
 impl TryFrom<RatchetPickle> for Ratchet {
     type Error = MegolmRatchetUnpicklingError;
 
     fn try_from(pickle: RatchetPickle) -> Result<Self, Self::Error> {
-        Ok(Ratchet {
```

Comment:
Need to remember that we need to take another look at this after the simplified pickling branch is merged.

Reply:
Added a TODO.